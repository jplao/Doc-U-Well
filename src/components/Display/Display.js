import React, { Component } from 'react';
import '../../index.scss';
import PropTypes from 'prop-types';

class Display extends Component {

  returnJsx (btnName, information) {
    return (
      <div className='display-container'>
        <button className='add-btn'> { btnName } <i className="far fa-plus-square"></i></button>
        <div className='card-area'> { information } </div>
      </div>
    )
  }

  render() {
    let url
    let btnName
    let information

    switch (this.props.type) {
      case 'profile' : 
        url = 'url to get profile'
        // this.props.fetchProfile(url)
        btnName = 'Update Profile'
        information = 'render Profile info on Card component, assign a key'
        return this.returnJsx(btnName, information)

      case 'appointments' : 
        url = 'url for get all appointments'
        // this.props.fetchAppointments(url)
        btnName = 'Add Appointment'
        information = 'map over Appointment and return Card component, spread in return info and assign a key'
        return this.returnJsx(btnName, information)

      case 'providers' : 
        url = 'url for get all providers'
        // this.props.fetchProviders(url)
        btnName = 'Add Provider'
        information = 'map over providers and return Card component, spread in return info and assign a key'
        return this.returnJsx(btnName, information)

      case 'insurance' : 
        url = 'url for get all insurance'
        // this.props.fetchInsurance(url)
        btnName = 'Add Insurance'
        information = 'map over insurance and return Card component, spread in return info and assign a key'
        return this.returnJsx(btnName, information)

      default:
        return ('should not hit default')
    }
  }
}

Display.propTypes = {
  type: PropTypes.string
}

export default Display;