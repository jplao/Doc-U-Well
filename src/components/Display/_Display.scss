.display-container {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 5fr;
  justify-items: center;
  align-items: start; 
  width: 100%;
  height: 100%;
  margin: 10px;
}

.add-btn {
  margin: 10px;
  padding: 10px;
  width: 70%;
  border-radius: 10px;
  position: relative;
  font-size: 1rem;
  color: #301e0a;
  outline: none;

  &:hover {
    background-color: #9ec65d;
    color: #fdf2ea;

  }
}

.card-area {
  margin: 10px;
  padding: 10px;
  overflow: auto;
}

.fa-plus-square {
  position: absolute;
  right: 10px;
  top: 10px;
}