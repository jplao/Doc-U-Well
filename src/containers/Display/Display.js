import React, { Component } from 'react';
import '../../index.scss';
import PropTypes from 'prop-types';
import Card from '../Card/Card'
import Schedule from '../../components/Schedule/Schedule'
import { connect } from 'react-redux';
import { apiKey } from '../../utils/apiKey'
import { getAllAppointmentsThunk } from '../../thunks/appointmentThunk/getAppointmentsThunk'
import { getAllProvidersThunk } from '../../thunks/providerThunk/getProvidersThunk'




class Display extends Component {
  constructor() {
    super()
    this.state = {
      toggleForm: false
    }
  }

  showForm () {
    this.setState({toggleForm: !this.state.toggleForm})
  }

  returnJsx (btnName, responseArray) {
    let info = responseArray.map(arrayItem => {
      return <Card {...arrayItem} type={this.props.type} key='arrayItem.name'/>
    })

    if(this.state.toggleForm) {
      return (
        <div className='display-container'>
          <button onClick={() => this.showForm()} className='add-btn'> Close Form <i className="far fa-plus-square"></i></button>
          <Schedule type={this.props.type}/>
          <div className='card-area'> { info } </div>
        </div>
      )
    } else {
      return(
        <div className='display-container'>
          <button onClick={() => this.showForm()} className='add-btn'> { btnName } <i className="far fa-plus-square"></i></button>
          <div className='card-area'> { info } </div>
        </div>
      )
    }
  }

  render () {
    let url
    let btnName
    let responseArray
    const { type, getAllProviders, provider} = this.props

    switch (type) {
      case 'profile' :
        url = 'url to get profile'
        // await this.props.fetchProfile(url)
        btnName = 'Add Profile'
        responseArray = [{blood: 'A+', height: '5,7', weight: '170', bps: '120', bpd: '80', hr: '60' }]
        return this.returnJsx(btnName, responseArray)

      case 'appointments' :
        url = 'url for get all appointments'
        // await this.props.fetchAppointments(url)
        btnName = 'Add Appointment'
        responseArray = [{date: 'Feb 14, 2019', time: '9:00 am', reason: 'Annual checkup', provider: 'Dr. Love', location: '124 Street Rd. Midlothian, VA 45459'}, {date: 'Jan 24, 2019', time: '7:00 am', reason: 'dentist', provider: 'Dr. FeelGood', location: '555 Street Rd. Denver, CO 88888'}, {date: 'June 4, 2019', time: '5:00 pm', reason: 'lab-work', provider: 'n/a', location: '14 Green Blvd. Beach, CA 00087'},{date: 'Feb 14, 2019', time: '9:00 am', reason: 'Annual checkup', provider: 'Dr. Love', location: '124 Street Rd. Midlothian, VA 45459'}, {date: 'Jan 24, 2019', time: '7:00 am', reason: 'dentist', provider: 'Dr. FeelGood', location: '555 Street Rd. Denver, CO 88888'}, {date: 'June 4, 2019', time: '5:00 pm', reason: 'lab-work', provider: 'n/a', location: '14 Green Blvd. Beach, CA 00087'}]
        return this.returnJsx(btnName, responseArray)

      case 'providers' :
        url = 'https://my-health-tracker.herokuapp.com/api/v1/providers'
                // await this.props.fetchProviders(url)
        getAllProviders(url)
        btnName = 'Add Provider'
        responseArray = provider
        return this.returnJsx(btnName, responseArray)

      case 'insurance' :
        url = 'url for get all insurance'
        // await this.props.fetchInsurance(url)
        btnName = 'Add Insurance'
        responseArray = [{card: 'Doctor', image: ':-)'}, {card: 'Dentist', image: ':-0'}]
        return this.returnJsx(btnName, responseArray)

      default:
        return ('hit display default')
    }
  }
}

const mapStateToProps = (state) => {
  return {
    appointment: state.appointments,
    provider: state.provider,
    insurance: state.insurance,
    profile: state.profile,
    isLoading: state.isLoading,
    errorMessage: state.errorMessage,
    }
}

const mapDispatchToProps = (dispatch) => ({
  getAllAppointments: (url => {
    dispatch(getAllAppointmentsThunk(url))
  }),
  getAllProviders: (url => {
    dispatch(getAllProvidersThunk(url))
  }),
  // getAllInsurance: (url => {
  //   dispatch(getAllInsuranceThunk(url))
  // }),
  // getAllProfile: (url => {
  //   dispatch(GetAllProfileThunk(url))
  // }),
})

Display.propTypes = {
  type: PropTypes.string
}

export default connect(mapStateToProps, mapDispatchToProps)(Display);
