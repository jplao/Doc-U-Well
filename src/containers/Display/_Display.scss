.display-container {
  display: grid;
  grid-template-columns: 1fr;
  justify-items: center;
  align-items: start; 
  width: 100%;
  height: 100%;
  margin: 10px;
}

.add-btn {
  margin-top: 10px;
  padding: 10px;
  width: 60%;
  border-radius: 10px;
  position: relative;
  font-size: 1rem;
  color: #301e0a;
  outline: none;

  &:hover {
    background-color: #9ec65d;
    color: #fdf2ea;
  }
}

.card-area {
  margin-bottom: 10px;
  padding: 10px;
  overflow: auto;
  width: 95%;
  height: 100%;
}

.fa-plus-square {
  position: absolute;
  right: 10px;
  top: 10px;
}