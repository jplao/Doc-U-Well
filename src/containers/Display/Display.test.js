import React from 'react';
import Display from './Display';
import { shallow } from 'enzyme';

describe('Display Container', () => {
  it('should match the screenshot', () => {
    const wrapper = shallow(<Display />)
    expect(wrapper).toMatchSnapshot()

  })
})
