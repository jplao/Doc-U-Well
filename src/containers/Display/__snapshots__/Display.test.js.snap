// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Container should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "hit display default",
  Symbol(enzyme.__nodes__): Array [
    "hit display default",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
